Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -2713,7 +2713,18 @@ private Task SchedulePip(PipId pipId, PipType pipType, RunnablePipObserver obser
             // Offload the execution of the pip to one of the queues in the PipQueue.
             // If it is a meta or SealDirectory pip and the PipQueue has started draining, then the execution will be inlined here!
             // Because it is not worth to enqueue the fast operations such as the execution of meta and SealDirectory pips.
-            var runnablePip = RunnablePip.Create(m_executePhaseLoggingContext, this, pipId, pipType, priority ?? GetPipPriority(pipId), m_executePipFunc);
+
+            ushort cpuUsageInPercent = m_scheduleConfiguration.UseHistoricalCpuUsageInfo ? RunningTimeTable[m_pipTable.GetPipSemiStableHash(pipId)].ProcessorsInPercents : (ushort)0;

```

Comment:
>ProcessorsInPercents [](start = 148, length = 20)

Amazing we had this already calculated but never actually utilized it...

Reply:
exactly. I started doing the changes in running time table, and then I noticed that we already have what we need there.